File: 74.js
function lengthOfLongestSubstring(s) {
    const lastIndex = new Map(); // char -> last seen index
    let left = 0;
    let maxLen = 0;

    for (let right = 0; right < s.length; right++) {
        const ch = s[right];
        if (lastIndex.has(ch) && lastIndex.get(ch) >= left) {
            left = lastIndex.get(ch) + 1;
        }
        lastIndex.set(ch, right);
        maxLen = Math.max(maxLen, right - left + 1);
    }

    return maxLen;
}

File: 74.py
def length_of_longest_substring(s: str) -> int:
    last_index = {}  # char -> last seen index
    left = 0
    max_len = 0

    for right, ch in enumerate(s):
        if ch in last_index and last_index[ch] >= left:
            # Move left just after the previous occurrence
            left = last_index[ch] + 1
        last_index[ch] = right
        max_len = max(max_len, right - left + 1)

    return max_len
